### PR TITLE
Diff two different tagged versions of the same run

### LINKED ISSUE #1346: It should be possible to diff two different tagged versions of the same run
It is only possible to ask for diff of two different runs but not two different tagged versions.
this features should be available from the
- [ ] command line with the format: `run_name:run_tag`
- [ ] GUI should use the same format as the command line `run_name:tag` 

Counting should be done like this see #1443:
 - non unique by default
 - set default filters to (unreviewed, confirmed, new, reopened, unresolved)
 - hash list based calculated based on dates detected, fixed

### PR DESCRIPTION
**Pull request by csordasmarton**
Closes #1346

### CODE REVIEW

File: libcodechecker/server/api/report_server.py [hunk 1]
```diff
@@ -410,6 +410,23 @@ def get_report_path_hash(report, files):
     return hashlib.md5(report_path_hash.encode()).hexdigest()
 
 
+def get_report_hashes_for_tags(session, run_ids, tag_ids):
```
[the remark this answers is not in the document]
Comment (gyorb) [reply libcodechecker/server/api/report_server.py]: We can return the hashes for the run_ids too but the function name does not reflect that.

Couldn't we find out the run_ids from the tag_ids or we use this function to get the hashes only for the run_ids too?

File: libcodechecker/server/api/report_server.py [hunk 2]
```diff
@@ -712,11 +729,18 @@ def getRunResults(self, run_ids, limit, offset, sort_types,
...
                                                         cmp_data)
                 if not diff_hashes:
                     # There is no difference.
                     return results
 
+                # If we are in diff mode and run tag is set in compare data,
+                # after diff hashes are calculated remove it from the
+                # report filter.
+                if cmp_data.runTag:
+                    report_filter.runTag = None
```
Comment (gyorb) [inline libcodechecker/server/api/report_server.py]: Could you describe it in more detail why do we need to remove the runTag from the filter?

File: libcodechecker/server/api/report_server.py [hunk 3]
```diff
@@ -909,11 +933,18 @@ def getRunResultCount(self, run_ids, report_filter, cmp_data):
             if cmp_data:
                 diff_hashes, run_ids = self._cmp_helper(session,
                                                         run_ids,
+                                                        report_filter,
                                                         cmp_data)
                 if not diff_hashes:
                     # There is no difference.
                     return 0
 
+                # If we are in diff mode and run tag is set in compare data,
```
Comment (gyorb) [inline libcodechecker/server/api/report_server.py]: Could you describe it a bit more why it is removed?

**Comment by gyorb** [comment]
Please resolve the conflict.

**Comment by csordasmarton** [comment]
@gyorb Done.

**Comment by gyorb** [comment]
Please resolve the conflicts

**Comment by csordasmarton** [comment]
@gyorb Done.

**Pull request merged**